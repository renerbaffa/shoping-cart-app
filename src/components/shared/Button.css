.Button {
  background-color: white;
  padding: 0px 18px;
}
