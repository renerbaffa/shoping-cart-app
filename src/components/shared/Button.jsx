import React from 'react';

/*
  Button component is an adapter.
  Imagine that in the future we move to react-bootstrap or material-ui
  In this case we only need to replase default button by the library button
*/
const Button = ({ children, ...other }) => (
  <button {...other}>
    {children}
  </button>
);

export default Button;
