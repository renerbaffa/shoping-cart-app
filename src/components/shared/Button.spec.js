import React from 'react';
import renderer from 'react-test-renderer';

import Button from './Button';

describe('<Button />', () => {
  it('should render correctly', () => {
    const buttonSnapshot =
      renderer
        .create(
          <Button className="className">
            Grid
          </Button>
        )
        .toJSON();
    expect(buttonSnapshot).toMatchSnapshot();
  });

  it('should apply selected classname', () => {
    const selectedButtonSnapshot =
      renderer
        .create(
          <Button className="className" selected>
            Grid
          </Button>
        ).toJSON();
      expect(selectedButtonSnapshot).toMatchSnapshot();
  });
});
