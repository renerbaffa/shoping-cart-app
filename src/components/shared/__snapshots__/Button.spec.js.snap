// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should apply selected classname 1`] = `
<button
  className="className Button Button-selected"
>
  Grid
</button>
`;

exports[`<Button /> should render correctly 1`] = `
<button
  className="className Button"
>
  Grid
</button>
`;
